lib/src/features/city_details/data/models/city_model.dart

```diff
@@ -0,0 +1,49 @@
+
+import 'package:emotion/src/features/city_details/data/models/city_provider_model.dart';
+import 'package:emotion/src/features/city_details/data/models/country_model.dart';
+import 'package:emotion/src/features/city_details/domain/entities/city.dart';
+
+class CityModel extends City {
+  const CityModel({
+    required int id,
+    required String name,
+    required String slug,
+    required String latitude,
+    required String longitude,
+    // ignore: non_constant_identifier_names
+    required List<dynamic> city_alternative_names,
+    required List<CityProviderModel> cityProviders,
+    required CountryModel country,
+    required List<dynamic> cityOpinions,
+  }) : super(
+          id: id,
+          name: name,
+          slug: slug,
+          latitude: latitude,
+          longitude: longitude,
+          // ignore: non_constant_identifier_names
+          city_alternative_names: city_alternative_names,
+          cityProviders: cityProviders,
+          country: country,
+          cityOpinions: cityOpinions,
+        );
+
+  factory CityModel.fromJson(Map<String, dynamic> json) {
```

Comment:
Personally, I like to use `json_serializable` package, which automatically generates `fromJson` and `toJson` implementations based on class properties. The `freezed` package does something similar, along with many others.

Reply:
I'll check the json_serializable, thanks for advice.